// This file was generated with the Protobuf generator tool.
// Dataset ID    : 23380
// Dataset Name  : cjs;
// Dataset URL   : https://www.openml.org/data/v1/download/1910442/cjs.arff
// Num. Columns  : 34
// Num. Rows     : 2796
// Target Feature: TR

// Beginning of Description of Dataset:
// **Author**: Dr. Fernando Camacho  
// **Source**: Unknown - 1995  
// **Please cite**: Camacho, F. and Arron, G. (1995)  Effects of the regulators paclobutrazol and flurprimidol on the growth of terminal sprouts formed on trimmed silver maple trees. Canadian Journal of Statistics 3(23).
// Data on tree growth used in the Case Study published in the September, 1995 issue of the Canadian Journal of Statistics. This data set was been provided by Dr. Fernando Camacho, Ontario Hydro Technologies, 800 Kipling Ave, Toronto Canada M3Z 5S4. It forms the basis of the Case Study in Data Analysis published in the Canadian Journal of Statistics, September 1995. It can be freely used for noncommercial purposes, as long as proper acknowledgement to the source and to the Canadian Journal of Statistics is made.
//

syntax = "proto3";

message Empty {
}

message Features {
	string   Tr                             = 1;
	string   Tree                           = 2;
	string   Br                             = 3;
	double   Tl                             = 4;
	uint32   In                             = 5;
	double   Internode_1                    = 6;
	double   Internode_2                    = 7;
	double   Internode_3                    = 8;
	double   Internode_4                    = 9;
	double   Internode_5                    = 10;
	double   Internode_6                    = 11;
	double   Internode_7                    = 12;
	double   Internode_8                    = 13;
	double   Internode_9                    = 14;
	double   Internode_10                   = 15;
	double   Internode_11                   = 16;
	double   Internode_12                   = 17;
	double   Internode_13                   = 18;
	double   Internode_14                   = 19;
	double   Internode_15                   = 20;
	double   Internode_16                   = 21;
	double   Internode_17                   = 22;
	double   Internode_18                   = 23;
	double   Internode_19                   = 24;
	double   Internode_20                   = 25;
	double   Internode_21                   = 26;
	double   Internode_22                   = 27;
	double   Internode_23                   = 28;
	double   Internode_24                   = 29;
	double   Internode_25                   = 30;
	double   Internode_26                   = 31;
	double   Internode_27                   = 32;
	double   Internode_28                   = 33;
	double   Internode_29                   = 34;
}

service get_next_row {
	 rpc get_next_row(Empty) returns(Features);
}

/*
//This code section can be directly used
//at the end of gen_next_row in the server.py file
//for OpenML dataset nr. 23380
        response.Tr                             = row[0]
        response.Tree                           = row[1]
        response.Br                             = row[2]
        response.Tl                             = row[3]
        response.In                             = numpy.uint32(row[4])
        response.Internode_1                    = row[5]
        response.Internode_2                    = row[6]
        response.Internode_3                    = row[7]
        response.Internode_4                    = row[8]
        response.Internode_5                    = row[9]
        response.Internode_6                    = row[10]
        response.Internode_7                    = row[11]
        response.Internode_8                    = row[12]
        response.Internode_9                    = row[13]
        response.Internode_10                   = row[14]
        response.Internode_11                   = row[15]
        response.Internode_12                   = row[16]
        response.Internode_13                   = row[17]
        response.Internode_14                   = row[18]
        response.Internode_15                   = row[19]
        response.Internode_16                   = row[20]
        response.Internode_17                   = row[21]
        response.Internode_18                   = row[22]
        response.Internode_19                   = row[23]
        response.Internode_20                   = row[24]
        response.Internode_21                   = row[25]
        response.Internode_22                   = row[26]
        response.Internode_23                   = row[27]
        response.Internode_24                   = row[28]
        response.Internode_25                   = row[29]
        response.Internode_26                   = row[30]
        response.Internode_27                   = row[31]
        response.Internode_28                   = row[32]
        response.Internode_29                   = row[33]
*/

